Makefile.include

```diff
@@ -727,8 +727,15 @@ else
   _LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $$(find $(BASELIBS:%.module=$(BINDIR)/%/) -name "*.o" 2> /dev/null | sort) $(ARCHIVES_GROUP) $(LINKFLAGS) $(LINKFLAGPREFIX)-Map=$(BINDIR)/$(APPLICATION).map
 endif # BUILDOSXNATIVE
 
+COMPILE_COMMANDS_PATH ?= $(RIOTBASE)/compile_commands.json
+COMPILE_COMMANDS_FLAGS ?= --clangd
+.PHONY: compile-commands
+
 ifeq ($(BUILD_IN_DOCKER),1)
 link: ..in-docker-container
+compile-commands: ..in-docker-container
+	@echo "Trying to patch paths in compile_commands.json: s!/data/riotbuild/riotbase!$(RIOTBASE)!g"
+	$(Q)sed -e "s!/data/riotbuild/riotbase!$(RIOTBASE)!g" -i "$(COMPILE_COMMANDS_PATH)"
```

Comment:
Wow, so, we create the compile commands in docker, with compiler and file paths pointing to the in-docker representation. then we fixup those paths.
How much sense does this make? The compile commands won't work outside docker, right?
Shouldn't the compile commands rather be fixed up to call `$(docker ...) $(actual compile command)`?

Reply:
I fear that I create the compile commands in docker for a target where I don't have the toolchain locally.
My editor picks up `compile-commands.json`, matches the path (which now points to the local file), and then tries to execute the `probably-xtensa-gcc` that the compile command entry specifies, but doesn't exist outside of docker -> executing the file fails.

Reply:
The `compile_commands.json` is something from the LLVM world. GCC compilers don't know what it is.

The main (and likely sole) use case is to have clangd knows where to search header files and what defines are present, so that linting and code compiletion in neovim / vim + ALE or in VS Code works correctly. The editor will need the paths from the editors point of view, not from within the docker container.

The patching of the `compile_commands json` is done once the docker is left, so any users in the docker image (such as rust) will read an un-patched version.